### PR TITLE
added controllers and dtos

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controllers/MovieController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    @PostMapping
+    public void add(@RequestBody MovieRequestDto movieRequestDto) {
+        movieService.add(movieMapper.fromRequestDto(movieRequestDto));
+    }
+
+    @GetMapping
+    public List<MovieResponseDto> getAll() {
+        return movieService.getAll()
+                .stream()
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/controllers/MovieController.java]:
```suggestion
        return movieService.getAll().stream()
```

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    @PostMapping
+    public void add(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(movieSessionMapper.fromRequestDto(movieSessionRequestDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableSessions(@RequestParam Long movieId,
+                                                              @RequestParam LocalDate date) {
+        return movieSessionService.findAvailableSessions(movieId, date)
+                .stream()
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]:
```suggestion
        return movieSessionService.findAvailableSessions(movieId, date).stream()
```

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @PostMapping
+    public void add(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(movieSessionMapper.fromRequestDto(movieSessionRequestDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableSessions(@RequestParam Long movieId,
+                                                              @RequestParam LocalDate date) {
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: Missing DateFormat annotation

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]
good job! made one comment below

File: src/main/java/com/dev/cinema/controllers/MovieController.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movies")
+public class MovieController {
+    public final MovieService movieService;
+    public final MovieMapper movieMapper;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/MovieController.java]: these fields should be private in all controllers and in mappers

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controllers/MovieController.java (hunk 4 above)
Comment (Martynice) [reply src/main/java/com/dev/cinema/controllers/MovieController.java]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**